### PR TITLE
Add hpu device support in storage/resize

### PR DESCRIPTION
**Pull request by jeejakp12**
Add hpu device to
 - In storage method resize_
 - is_supported_device for fsdp
 - for storage add hpu device support

Fixes #ISSUE_NUMBER

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225

### CODE REVIEW

File: torch/storage.py [hunk 1]
```diff
@@ -1181,7 +1181,7 @@ def _get_legacy_storage_class(self):
 
         storage_name = _dtype_to_storage_type_map()[self.dtype]
 
-        if self.device.type not in ['cpu', 'cuda', torch._C._get_privateuse1_backend_name()]:
+        if self.device.type not in ['cpu', 'cuda', "hpu", torch._C._get_privateuse1_backend_name()]:
```
Comment (mikaylagawarecki) [inline torch/storage.py:1184]:
```suggestion
        if self.device.type not in ['cpu', 'cuda', 'hpu', torch._C._get_privateuse1_backend_name()]:
```

**Review by mikaylagawarecki — DISMISSED** [review]

**Comment by mikaylagawarecki** [comment]
fyi, this might landrace with https://github.com/pytorch/pytorch/pull/119671 which migrated the `THPStorage_resize_` logic, I would recommend rebasing after that lands

**Comment by jeejakp12** [comment]
> fyi, this might landrace with #119671 which migrated the `THPStorage_resize_` logic, I would recommend rebasing after that lands

yes, rebased and pushed the chnages

**Comment by jeejakp12** [comment]
@mikaylagawarecki rebased and pushed the patch.

**Review by mikaylagawarecki — APPROVED** [review]

**Pull request closed without merging**